src/components/subscriptions/SubscriptionRoutes.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import moment from 'moment';
+import {
+  useHistory,
+  useRouteMatch,
+  Switch,
+  Route,
+} from 'react-router-dom';
+import { Tabs, Tab } from '@edx/paragon';
+
+import MultipleSubscriptionsPage from './MultipleSubscriptionsPage';
+import SubscriptionDetailPage from './SubscriptionDetailPage';
+import SubscriptionSubsidyRequests from './SubscriptionSubsidyRequests';
+import { ROUTE_NAMES } from '../EnterpriseApp/constants';
+import { features } from '../../config';
+
+const SubscriptionPlanRoutes = ({ enterpriseSlug }) => {
+  const multipleSubsCreateActions = (subscription) => {
+    const now = moment();
+    const isScheduled = now.isBefore(subscription.startDate);
+    const isExpired = now.isAfter(subscription.expirationDate);
+    const buttonText = `${isExpired ? 'View' : 'Manage'} learners`;
+    const buttonVariant = isExpired ? 'outline-primary' : 'primary';
+
+    const actions = [];
+
+    if (!isScheduled) {
+      actions.push({
+        variant: buttonVariant,
+        to: `/${enterpriseSlug}/admin/${ROUTE_NAMES.subscriptionManagement}/${subscription.uuid}`,
+        buttonText,
+      });
+    }
+
+    return actions;
+  };
+
+  return (
+    <>
+      <Route
+        path={`/:enterpriseSlug/admin/${ROUTE_NAMES.subscriptionManagement}`}
+        render={routeProps => (
+          <MultipleSubscriptionsPage
+            {...routeProps}
+            createActions={multipleSubsCreateActions}
+          />
+        )}
+        exact
+      />
+      <Route
+        path={`/:enterpriseSlug/admin/${ROUTE_NAMES.subscriptionManagement}/:subscriptionUUID`}
+        component={SubscriptionDetailPage}
+        exact
+      />
+    </>
+  );
+};
+
+SubscriptionPlanRoutes.propTypes = {
+  enterpriseSlug: PropTypes.string.isRequired,
+};
+
+const SubscriptionRoutes = ({ enterpriseSlug }) => {
+  const isTabsFeatureEnabled = !!features.FEATURE_BROWSE_AND_REQUEST;
+
+  const routesByTabKey = {
+    requests: `/${enterpriseSlug}/admin/${ROUTE_NAMES.subscriptionManagement}/requests`,
+    default: `/${enterpriseSlug}/admin/${ROUTE_NAMES.subscriptionManagement}`,
+  };
+  const history = useHistory();
+  const requestsTabMatch = useRouteMatch(routesByTabKey.requests);
+  const initialTabKey = requestsTabMatch ? 'requests' : 'default';
+  const [tabKey, setTabKey] = useState(initialTabKey);
```

Comment:
See comment on `CodeManagementRoutes` that allows us to remove `useSate` and `useEffect` and simplify this file.

Reply:
Updated!